qp_klp/Amplicon.py

```diff
@@ -81,23 +83,43 @@ def quality_control(self):
                 new_path = join(output_folder, basename(raw_fastq_file))
                 copyfile(raw_fastq_file, new_path)
 
-    def generate_reports(self, input_file_path):
-        super()._generate_reports(self.pipeline.mapping_file)
+    def generate_reports(self):
+        super()._generate_reports()
         return None  # amplicon doesn't need project names
 
+    def generate_touched_studies(self, qclient):
+        for study_id in self.prep_file_paths:
+            for prep_file_path in self.prep_file_paths[study_id]:
+                metadata = Step.parse_prep_file(prep_file_path)
+                if 'target_gene' in metadata[list(metadata.keys())[0]]:
+                    tg = metadata[list(metadata.keys())[0]]['target_gene']
+                    data_type = None
+                    for key in Step.AMPLICON_SUB_TYPES:
+                        if key in tg:
+                            data_type = key
+
+                    if data_type is None:
+                        raise ValueError("A valid data-type could not be "
+                                         "derived from target_gene column")
+
+            super()._generate_touched_studies(qclient, data_type)
+
     def generate_prep_file(self):
         config = self.pipeline.configuration['seqpro']
 
         seqpro_path = config['seqpro_path'].replace('seqpro', 'seqpro_mf')
 
+        projects = self.pipeline.get_project_info()
+        project_names = [x['project_name'] for x in projects]
```

Comment:
Can you merge these 2 lines in a single loop?
`project_names = [x['project_name'] for x in project_names = [x['project_name'] for x in projects]]`